src/jit/importer.cpp

```diff
@@ -12549,7 +12551,20 @@ void Compiler::impImportBlockCode(BasicBlock* block)
                             op1->AsCall()->gtCallType == CT_HELPER)
 #endif // UNIX_AMD64_ABI
                         {
-                            op1 = impGetStructAddr(op1, clsHnd, (unsigned)CHECK_SPILL_ALL, false);
+                            // If the value being produced comes from loading
+                            // via an underlying address, just null check the address.
+                            if (op1->OperIs(GT_FIELD, GT_IND, GT_OBJ))
```

Comment:
Hmm, any volatile concerns?

Reply:
Hmm, maybe so. Thanks for bringing that up.

A quick glance suggests the jit won't even record volatile prefixes for ldobj. Need to dig in some more.

Reply:
Ah, this case is for structs. It's not clear to me if volatile applies to structs. ECMA-335 doesn't seem to say that volatile is not allowed on structs but I don't think I've ever seen code using that.

But then it looks like the importer already drops popped volatile reads - #6172